pkg/volume/local/local.go

```diff
@@ -38,6 +38,10 @@ import (
 	"k8s.io/kubernetes/pkg/volume/validation"
 )
 
+const (
+	defaultFSType = "ext4"
```

Comment:
Why do we need to make this assumption?  In general Linux can auto-detect filesystems, right?  why not make the default be best-effort auto-detect, and try to avoid encouraging people to make assumptions.

This sort of stuff is also hard to ferret out as Windows support becomes more of a Real Thing (@kubernetes/sig-windows-misc )

```
# cd /tmp

# dd if=/dev/zero of=loopfile bs=$((1024*1024)) count=64
64+0 records in
64+0 records out
67108864 bytes (67 MB, 64 MiB) copied, 0.0661057 s, 1.0 GB/s

# mkfs.xfs ./loopfile 
meta-data=./loopfile             isize=512    agcount=4, agsize=4096 blks
         =                       sectsz=512   attr=2, projid32bit=1
         =                       crc=1        finobt=1, sparse=0, rmapbt=0, reflink=0
data     =                       bsize=4096   blocks=16384, imaxpct=25
         =                       sunit=0      swidth=0 blks
naming   =version 2              bsize=4096   ascii-ci=0 ftype=1
log      =internal log           bsize=4096   blocks=855, version=2
         =                       sectsz=512   sunit=0 blks, lazy-count=1
realtime =none                   extsz=4096   blocks=0, rtextents=0

# mkdir mnt

# mount -oloop ./loopfile ./mnt

# mount | grep loopfile
/tmp/loopfile on /tmp/mnt type xfs (rw,relatime,attr2,inode64,noquota)

# umount ./mnt
```

Reply:
For the case of format-on-demand, (assuming that's what really drives this) perhaps it should be "whatever the node in question thinks is best"?  Can we make that something the host defines?  I feel like we've had this conversation before and I lost it then, but I forget why, so I'll happily take another tilt at it..

Reply:
The primary purpose of this field is for the format-on-demand case, not really for identifying the filesystem type on existing volumes.

We could remove the fstype field and just format it with "whatever the host wants", but how can we support situations where the user wants something different?